### PR TITLE
Made internal links and smart content show unpublished nodes

### PR DESCRIPTION
**Pull request by marcelmoosbrugger**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | no
| Enhancement? | yes
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes https://github.com/sulu/sulu/issues/821
| Related issues/PRs | https://github.com/massiveart/husky/pull/698
| License | MIT
| Documentation PR | none

#### What's in this PR?

The backend was adapted so also unpublished and pages which have a draft version get returned. In the frontend, the smart-content now is able to visualize the different states of the pages it displays. Moreover a husky version is included which visualizes the page states for the internal links.

### CODE REVIEW

**Comment by marcelmoosbrugger** [comment]
@danrot. If the tests pass, this PR is ready for reviewing.

File: src/Sulu/Component/Content/Tests/Functional/SmartContent/SmartContentQueryBuilderTest.php [hunk 1]
```diff
@@ -1114,16 +1114,22 @@ public function testShadow()
...
-        $this->assertEquals('Daniel', $result[0]['title']);
-        $this->assertEquals('/team/johannes', $result[1]['url']);
-        $this->assertEquals('Johannes DE', $result[1]['title']);
+        $this->assertEquals(3, count($result));
+        $this->assertEquals('/team/thomas', $result[0]['url']);
+        $this->assertEquals('Thomas', $result[0]['title']);
+        $this->assertEquals('/team/daniel', $result[1]['url']);
+        $this->assertEquals('Daniel', $result[1]['title']);
+        $this->assertEquals('/team/johannes', $result[2]['url']);
+        $this->assertEquals('Johannes DE', $result[2]['title']);
```
Comment (danrot) [inline src/Sulu/Component/Content/Tests/Functional/SmartContent/SmartContentQueryBuilderTest.php]: Don't you also want to test the values of your newly added fields here?

File: CHANGELOG.md [hunk 2]
```diff
@@ -50,8 +51,6 @@ CHANGELOG for Sulu
     * BUGFIX      #2590 [CoreBundle]          Clear symfony cache before system collection initialization
     * BUGFIX      #2596 [PreviewBundle]       Fixed preview for prod environment
     * FEATURE     #2515 [ContentBundle]       Added unpublishing functionality for pages
-    * ENHANCEMENT #2604 [ContentBundle]       fixed publishing on excerpt tab and add excerpt js extension
-    * BUGFIX      #2586 [AdminBundle]         fixed behat tests
```
Comment (danrot) [inline CHANGELOG.md]: How can these lines be removed?

File: src/Sulu/Bundle/ContentBundle/Resources/public/js/components/internal-links/main.js [hunk 3]
```diff
@@ -44,12 +46,25 @@ define([], function() {
...
                 return [
+                    '<span class="icons">' + icons + '</span>',
                     '<a href="#" data-id="', id, '" class="link">',
                     '    <span class="value">', value, '</span>',
                     '</a>'
                 ].join('');
+            },
+
+            icons: {
+                notPublishedState: function(title) {
```
Comment (danrot) [inline src/Sulu/Bundle/ContentBundle/Resources/public/js/components/internal-links/main.js]: I know these variables are called like this in the API, but why can't we at least use better names here? `onlyDraft` and `publishedAndDraft` sound already better, don't they? Or would it maybe even be better to split the two templates completely?

File: src/Sulu/Bundle/ContentBundle/Resources/public/js/components/smart-content/main.js [hunk 4]
```diff
@@ -302,6 +306,14 @@ define(['services/husky/util'], function(util) {
                     '    <input type="text" value="<%= limitResult %>" class="limit-to form-element"<%= disabled %>/>',
                     '</div>'
                 ].join('')
+            },
+
+            icons: {
+                notPublishedState: '<span class="not-published-state-icon" title="<%= title %>"/>',
```
Comment (danrot) [inline src/Sulu/Bundle/ContentBundle/Resources/public/js/components/smart-content/main.js]: Same question as in the other component.

File: CHANGELOG.md (hunk 2 above)
Comment (marcelmoosbrugger) [inline CHANGELOG.md]: I removed them. They were duplicates (also in dev-develop)

File: src/Sulu/Component/Content/Mapper/ContentMapper.php [hunk 5]
```diff
@@ -729,6 +729,7 @@ private function rowToArray(Row $row, $locale, $webspaceKey, $fields, $onlyPubli
             'changed' => $document->getChanged(),
             'changer' => $document->getChanger(),
             'created' => $document->getCreated(),
+            'publishedState' => $nodeState === WorkflowStage::PUBLISHED,
```
Comment (danrot) [inline src/Sulu/Component/Content/Mapper/ContentMapper.php]: Why did you have to touch this file? Is there still a dependency to it in the scope of this PR?

**Pull request merged**